linux/just_files/just_install_functions.bsh

```diff
@@ -627,6 +627,103 @@ function pipenv-install()
 }
 
+#**
+# .. function:: vsbuild-install
+#
+# Install Visual Studio 2022 Build tools.
+#
+# :Arguments: * [``--installer {INSTALLER}``] - VS installer (default= ``${VS_BUILD_TOOLS_INSTALLER:-}``)
+#             * [``--download``] - Download VS installer and enables online mode
+#             * [``--channel``] - Channel URI. See [here](https://stackoverflow.com/a/78098290/4166604) to figure out a specific channel URI. E.g. ``https://aka.ms/vs/17/release.ltsc.17.0/channel``.
+#             * [``--add``] - Add additional components
+#
+# Additional components can be added; the intent is to specify additional MSVC versions.
+#
+# - 2022 - ``Microsoft.VisualStudio.Component.VC.Tools.x86.x64`` (always added, listed just for reference)
+# - 2017 - ``Microsoft.VisualStudio.Component.VC.v141.x86.x64``
+# - 2015 - ``Microsoft.VisualStudio.Component.VC.140``
+#
+# 2019 is more difficult. Every version has a version number in the component name. E.g.:
+# - ``Microsoft.VisualStudio.Component.VC.14.29.16.11.x86.x64``
+# - ``Microsoft.VisualStudio.Component.VC.14.30.17.0.x86.x64``
+# - ``Microsoft.VisualStudio.Component.VC.14.31.17.1.x86.x64``
+# - ...
+# - ``Microsoft.VisualStudio.Component.VC.14.40.17.10.x86.x64``
+# - ``Microsoft.VisualStudio.Component.VC.14.41.17.11.x86.x64``
+#
+# Similar to retrieving the channel URI, this 2019 list can be retrieved from the ``Catalog.json`` file:
+#
+# .. code:
+#
+#    jq . Catalog.json | \grep -i '"id": "Microsoft.VisualStudio.Component.VC.14.*.x86.x64"'
+#
+# Specifying the installer can be used for offline installations or to specify which version gets installed using [fixed version installers](https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-history#fixed-version-bootstrappers) instead of setting the Channel URI.
+#
+# 1. Download https://aka.ms/vs/17/release/vs_BuildTools.exe (or a fixed version installer)
+# 2. (For offline setup) ``vs_BuildTools.exe --layout C:\myLayoutDir --add Microsoft.VisualStudio.Component.VC.Tools.x86.x64 --lang en-US``
+#    - Transfer ``C:\myLayoutDir`` ...
+# 3. Now you can use ``--installer C:\myLayoutDir\vs_BuildTools.exe``
+#
+# .. note::
+#
+#    This will prompt UAC for required admin privileges.
+#**
+function vsbuild-install()
+{
+  if [ "${OS-}" != "Windows_NT" ]; then
+    echo "This is a windows only feature" &> 2
+    local JUST_IGNORE_EXIT_CODES=1
+    return 1
+  fi
+
+  local allow_download=0
+  local VS_BUILD_TOOLS_INSTALLER=${VS_BUILD_TOOLS_INSTALLER-}
+  local additional_packages=()
+
+  local channel_uri # Latest is https://aka.ms/vs/17/release/channel
+
+  parse_args installer_extra_args \
+      --installer VS_BUILD_TOOLS_INSTALLER: \
+      --download allow_download \
+      --add +additional_packages: \
+      --channel channel_uri: \
+      -- ${@+"${@}"}
+
+  # If the installer is not specified or found, and downloads are allowed
+  if [ ! -f "${VS_BUILD_TOOLS_INSTALLER}" ] && [ "${allow_download}" != "0" ]; then
```

Comment:
Throw a meaningful error if `allow_download` is disabled and `$VS_BUILD_TOOLS_INSTALLER` was not specified?